bitcoin/src/sign_message.rs

```diff
@@ -108,7 +108,25 @@ mod message_signing {
             serialized
         }
 
+        /// Creates a `MessageSignature` from a fixed-length array.
+        pub fn from_byte_array(
+            bytes: &[u8; 65],
+        ) -> Result<MessageSignature, MessageSignatureError> {
+            // We just check this here so we can safely subtract further.
+            if bytes[0] < 27 {
+                return Err(MessageSignatureError::InvalidEncoding(
+                    secp256k1::Error::InvalidRecoveryId,
+                ));
+            }
+            let recid = RecoveryId::from_i32(((bytes[0] - 27) & 0x03) as i32)?;
+            Ok(MessageSignature {
+                signature: RecoverableSignature::from_compact(&bytes[1..], recid)?,
+                compressed: ((bytes[0] - 27) & 0x04) != 0,
+            })
+        }
```

Comment:
It'd be nice to call into this from `from_slice` to ensure the implementations stay consistent even if it's just for the time of deprecation. It's also possible the diff will be easier to understand.

Reply:
Thanks, I have changed it to call `from_byte_array` from `from_slice`